planning/class_constructor/implementation_plan.md

```
@@ -0,0 +1,1066 @@
+the inferred default apply or annotate use sites explicitly.
+
+Update snapshot fixtures: `UPDATE_SNAPS=true go test ./...`.
+
+### 4.4 Parser Removal
+
+**File:** `internal/parser/decl.go`
+
+Delete the `if token.Type == OpenParen { … params = parseDelimSeq(…) }`
+block (currently around line 156). Encountering `(` after the class name
+(and optional type params) now reports `"primary constructors are no
+longer supported; declare an explicit `constructor` block in the class
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)